### PR TITLE
[doc-complete] #16203 Added doc to fix issue with missing client instructions for batch index acknowledgment

### PR DESCRIPTION
**Pull request by devinbost**
This change is documented here: https://github.com/apache/pulsar/issues/16203

I also made a minor spelling correction on that page.

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -174,16 +189,16 @@ Messages are received from [brokers](reference-terminology.md#broker) either syn
 
 Client libraries provide listener implementation for consumers. For example, the [Java client](client-libraries-java) provides a {@inject: javadoc:MesssageListener:/client/org/apache/pulsar/client/api/MessageListener} interface. In this interface, the `received` method is called whenever a new message is received.
 
-### Acknowledgement
+### acknowledgment
```
Comment (momo-jun) [inline site2/docs/concepts-messaging.md:192]:
```suggestion
### Acknowledgment
```
The first letter should be capitalized in a heading.

File: site2/docs/concepts-messaging.md [hunk 2]
```diff
@@ -279,11 +294,11 @@ If batching is enabled, all messages in one batch are redelivered to the consume
 
 :::
 
-### Acknowledgement timeout
+### acknowledgment timeout
```
Comment (momo-jun) [inline site2/docs/concepts-messaging.md:297]:
```suggestion
### Acknowledgment timeout
```

File: site2/docs/concepts-messaging.md [hunk 3]
```diff
@@ -876,7 +891,7 @@ There are diverse system topics depending on namespaces. The following table out
 | host/port | `heartbeat` | Persistent | 1 | Heartbeat detection |
 | User-defined-ns | [`__change_events`](concepts-multi-tenancy.md#namespace-change-events-and-topic-level-policies) | Persistent | Default 4 | Topic events |
 | User-defined-ns | `__transaction_buffer_snapshot` | Persistent | One per namespace | Transaction buffer snapshots |
-| User-defined-ns | `${topicName}__transaction_pending_ack` | Persistent | One per every topic subscription acknowledged with transactions | Acknowledgements with transactions |
+| User-defined-ns | `${topicName}__transaction_pending_ack` | Persistent | One per every topic subscription acknowledged with transactions | acknowledgments with transactions |
```
Comment (momo-jun) [inline site2/docs/concepts-messaging.md:894]:
```suggestion
| User-defined-ns | `${topicName}__transaction_pending_ack` | Persistent | One per every topic subscription acknowledged with transactions | Acknowledgments with transactions |
```

File: site2/docs/concepts-messaging.md [hunk 4]
```diff
@@ -903,8 +918,8 @@ Apache Pulsar avoids these and other message delivery failures using at-least-on
 
 To utilize message redelivery, you need to enable this mechanism before the broker can resend the unacknowledged messages in Apache Pulsar client. You can activate the message redelivery mechanism in Apache Pulsar using three methods. 
 
-- [Negative Acknowledgment](#negative-acknowledgement)
-- [Acknowledgement Timeout](#acknowledgement-timeout)
+- [Negative Acknowledgment](#negative-acknowledgment)
+- [acknowledgment Timeout](#acknowledgment-timeout)
```
Comment (momo-jun) [inline site2/docs/concepts-messaging.md:922]:
```suggestion
- [Acknowledgment Timeout](#acknowledgment-timeout)
```

**Pull request merged**